sycl/test-e2e/BFloat16/bfloat16_builtins.cpp

```diff
@@ -1,5 +1,6 @@
+// DEFINE: %{mathflags} = %if cl_options %{/clang:-fno-fast-math%} %else %{-fno-fast-math%}
 // REQUIRES: aspect-ext_oneapi_bfloat16_math_functions
-// RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} %if any-device-is-cuda %{ -Xsycl-target-backend --cuda-gpu-arch=sm_80 %} %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} %if any-device-is-cuda %{ -Xsycl-target-backend --cuda-gpu-arch=sm_80 %} %s -o %t.out %{mathflags}
```

Comment:
@JackAKirk  - I'm working with this test and on our slightly older shared CUDA dev machines `sm_80` gets rejected.  But with `sm_75` the test both compiles and behaves as expected.   

Can I just switch this to `sm_75` ? Or, better yet, given that we have a `REQUIRES: aspect-ext_oneapi_bfloat16_math_functions` in this test, can the whole `%if any-device-is-cuda ... %}` block be removed?

Reply:
The test behaves differently depending on whether it is compiled for sm_xx>=sm_80 or not:
- sm_80 and above uses some native bfloat16 math instructions
- below sm_80 always uses generic impls
- 
So I set it to compile with sm_80 flag because this is what the CI has, and allows testing of the native impls.

It is possible to remove the arch flag, and it is probably the best thing to do now that bfloat16 is generically supported, to avoid confusion. Unfortunately this means that the native impls won't be tested automatically via the CI. But I suppose we could test this in release testing.

`ext_oneapi_bfloat16_math_functions` is really an artifact of earlier times when bfloat16 was not generically implemented for all devices: I think it should be removed.

Thanks

Reply:
Given that, I think I'll have it test both and add a comment summarizing what you just said.